packages/in_app_purchase/in_app_purchase/README.md

```diff
@@ -109,53 +110,72 @@ class _MyAppState extends State<MyApp> {
 
 Here is an example of how to handle purchase updates:
 
+<?code-excerpt "example/lib/main.dart (HandlePurchase)"?>
 ```dart
-void _listenToPurchaseUpdated(List<PurchaseDetails> purchaseDetailsList) {
-  purchaseDetailsList.forEach((PurchaseDetails purchaseDetails) async {
+Future<void> _listenToPurchaseUpdated(
+    List<PurchaseDetails> purchaseDetailsList) async {
+  for (final PurchaseDetails purchaseDetails in purchaseDetailsList) {
     if (purchaseDetails.status == PurchaseStatus.pending) {
-      _showPendingUI();
+      showPendingUI();
     } else {
       if (purchaseDetails.status == PurchaseStatus.error) {
-        _handleError(purchaseDetails.error!);
+        handleError(purchaseDetails.error!);
       } else if (purchaseDetails.status == PurchaseStatus.purchased ||
-                 purchaseDetails.status == PurchaseStatus.restored) {
-        bool valid = await _verifyPurchase(purchaseDetails);
+          purchaseDetails.status == PurchaseStatus.restored) {
+        final bool valid = await _verifyPurchase(purchaseDetails);
         if (valid) {
-          _deliverProduct(purchaseDetails);
+          unawaited(deliverProduct(purchaseDetails));
         } else {
           _handleInvalidPurchase(purchaseDetails);
+          return;
+        }
+      }
+      if (Platform.isAndroid) {
+        if (!_kAutoConsume && purchaseDetails.productID == _kConsumableId) {
```

Comment:
Adding auto-consume and consumables to this example makes it much longer; I prefer the more targeted code we had before for an inline README example.

If someone wants more complex use cases, the example app is generally the right place for those examples, with the README doing short demonstrations of the most common cases.